### PR TITLE
Clean up gcodecanvas

### PR DESCRIPTION
**Pull request by johnbenweber**
I've written a gcode parsing class but was having some hiccups implementing it. I've noticed that some of the code could use a good clean so I thought I would get that out of the way first. First victim is gcodeCanvas.py. Removed some repeated code, renamed some functions for consistency and removed a (pointless?) redirect function. Added description comments to function strings. Renamed some variables and deleted some unused variables. Deleted 'D' command draw instructions because they are misleading since they currently do nothing.

### CODE REVIEW

File: UIElements/gcodeCanvas.py [hunk 1]
```diff
@@ -54,25 +49,21 @@ def initialize(self):
...
-        
-        #self.targetIndicator.color = (1,0,0)
-        
+
         tempViewMenu = ViewMenu()
         tempViewMenu.setUpData(self.data)
         tempViewMenu.reloadGcode()
     
-    def centerScatter(self, *args):
+    def centerCanvas(self, *args):
```
Comment (BarbourSmith) [inline UIElements/gcodeCanvas.py]: good names

File: UIElements/gcodeCanvas.py [hunk 2]
```diff
@@ -247,36 +188,48 @@ def drawG3(self,gCodeLine):
...
+
+    def updateGcode(self, *args):
+        '''
+        
+        updateGcode parses the gcode commands and calls the appropriate drawing function for the 
+        specified command. 
     
-    #This draws the gcode on the canvas.
-    def drawgcode(self):
-        xnow = 800.0
```
Comment (BarbourSmith) [inline UIElements/gcodeCanvas.py]: These are a hold over from the version of this software from 3+ years ago 🤢

File: UIElements/gcodeCanvas.py [hunk 3]
```diff
@@ -287,7 +240,7 @@ def drawgcode(self):
             #self.canv.create_text(xnow + 200, ynow - 50, text = errorText, fill = "red")
         
         
-        while i < len(self.data.gcode) and i < 8000: #maximum of 8,000 lines are drawn on the screen at a time
+        for i in range(0, min(len(self.data.gcode),8000)): #maximum of 8,000 lines are drawn on the screen at a time
```
Comment (BarbourSmith) [inline UIElements/gcodeCanvas.py]: ❤️

**Comment by BarbourSmith** [comment]
Making things cleaner is always very welcome! Great work. 👍 👍 👍 

I think I found a file which doesn't seem to render right on the new version. It should be a complete octagon circle thing, but it seems like only the first three sides render.

Here's the code:

`(Generated by PartKam Version 0.05)

G20 G90 G40

(CAUSE)
G0 Z0.25
T0 M6
G17
M3
G0 X1.3062 Y7.0303
G1 Z-0.05 F30
G1 X-1.6542 Y7.0316 F15
G3 X-1.7625 Y6.9692 I-0.0001 J-0.125
G1 X-2.9644 Y4.8899
G1 X-5.3915 Y4.8881
G3 X-5.4996 Y4.8257 I0.0001 J-0.125
G1 X-6.9785 Y2.2675
G3 X-6.9953 Y2.2049 I0.1082 J-0.0626
G3 X-6.9785 Y2.1424 I0.125 J0
G1 X-5.7791 Y0.0652
G1 X-6.9912 Y-2.0343
G3 X-7.0079 Y-2.0968 I0.1083 J-0.0625
G3 X-6.9912 Y-2.1593 I0.125 J0
G1 X-5.5121 Y-4.7238
G3 X-5.4039 Y-4.7863 I0.1083 J0.0625
G1 X-3.0014 Y-4.7876
G1 X-1.7718 Y-6.8577
G3 X-1.6655 Y-6.9189 I0.1075 J0.0638
G1 X1.2743 Y-6.9463
G3 X1.3837 Y-6.8838 I0.0012 J0.125
G1 X2.583 Y-4.8064
G1 X5.01 Y-4.8046
G3 X5.1182 Y-4.7421 I-0.0001 J0.125
G1 X6.5969 Y-2.1807
G3 X6.6136 Y-2.1182 I-0.1083 J0.0625
G3 X6.5968 Y-2.0556 I-0.125 J0
G1 X5.3977 Y0.0184
G1 X6.6097 Y2.1179
G3 X6.6265 Y2.1804 I-0.1083 J0.0625
G3 X6.6098 Y2.2428 I-0.125 J0
G1 X5.1336 Y4.8058
G3 X5.0253 Y4.8684 I-0.1083 J-0.0624
G1 X2.6266 Y4.8684
G1 X1.4144 Y6.9678
G3 X1.3062 Y7.0303 I-0.1083 J-0.0625
G1 Z-0.1 F30
G1 X-1.6542 Y7.0316 F15
G3 X-1.7625 Y6.9692 I-0.0001 J-0.125
G1 X-2.9644 Y4.8899
G1 X-5.3915 Y4.8881
G3 X-5.4996 Y4.8257 I0.0001 J-0.125
G1 X-6.9785 Y2.2675
G3 X-6.9953 Y2.2049 I0.1082 J-0.0626
G3 X-6.9785 Y2.1424 I0.125 J0
G1 X-5.7791 Y0.0652
G1 X-6.9912 Y-2.0343
G3 X-7.0079 Y-2.0968 I0.1083 J-0.0625
G3 X-6.9912 Y-2.1593 I0.125 J0
G1 X-5.5121 Y-4.7238
G3 X-5.4039 Y-4.7863 I0.1083 J0.0625
G1 X-3.0014 Y-4.7876
G1 X-1.7718 Y-6.8577
G3 X-1.6655 Y-6.9189 I0.1075 J0.0638
G1 X1.2743 Y-6.9463
G3 X1.3837 Y-6.8838 I0.0012 J0.125
G1 X2.583 Y-4.8064
G1 X5.01 Y-4.8046
G3 X5.1182 Y-4.7421 I-0.0001 J0.125
G1 X6.5969 Y-2.1807
G3 X6.6136 Y-2.1182 I-0.1083 J0.0625
G3 X6.5968 Y-2.0556 I-0.125 J0
G1 X5.3977 Y0.0184
G1 X6.6097 Y2.1179
G3 X6.6265 Y2.1804 I-0.1083 J0.0625
G3 X6.6098 Y2.2428 I-0.125 J0
G1 X5.1336 Y4.8058
G3 X5.0253 Y4.8684 I-0.1083 J-0.0624
G1 X2.6266 Y4.8684
G1 X1.4144 Y6.9678
G3 X1.3062 Y7.0303 I-0.1083 J-0.0625
G1 Z-0.15 F30
G1 X-1.6542 Y7.0316 F15
G3 X-1.7625 Y6.9692 I-0.0001 J-0.125
G1 X-2.9644 Y4.8899
G1 X-5.3915 Y4.8881
G3 X-5.4996 Y4.8257 I0.0001 J-0.125
G1 X-6.9785 Y2.2675
G3 X-6.9953 Y2.2049 I0.1082 J-0.0626
G3 X-6.9785 Y2.1424 I0.125 J0
G1 X-5.7791 Y0.0652
G1 X-6.9912 Y-2.0343
G3 X-7.0079 Y-2.0968 I0.1083 J-0.0625
G3 X-6.9912 Y-2.1593 I0.125 J0
G1 X-5.5121 Y-4.7238
G3 X-5.4039 Y-4.7863 I0.1083 J0.0625
G1 X-3.0014 Y-4.7876
G1 X-1.7718 Y-6.8577
G3 X-1.6655 Y-6.9189 I0.1075 J0.0638
G1 X1.2743 Y-6.9463
G3 X1.3837 Y-6.8838 I0.0012 J0.125
G1 X2.583 Y-4.8064
G1 X5.01 Y-4.8046
G3 X5.1182 Y-4.7421 I-0.0001 J0.125
G1 X6.5969 Y-2.1807
G3 X6.6136 Y-2.1182 I-0.1083 J0.0625
G3 X6.5968 Y-2.0556 I-0.125 J0
G1 X5.3977 Y0.0184
G1 X6.6097 Y2.1179
G3 X6.6265 Y2.1804 I-0.1083 J0.0625
G3 X6.6098 Y2.2428 I-0.125 J0
G1 X5.1336 Y4.8058
G3 X5.0253 Y4.8684 I-0.1083 J-0.0624
G1 X2.6266 Y4.8684
G1 X1.4144 Y6.9678
G3 X1.3062 Y7.0303 I-0.1083 J-0.0625
G1 Z-0.2 F30
G1 X-1.6542 Y7.0316 F15
G3 X-1.7625 Y6.9692 I-0.0001 J-0.125
G1 X-2.9644 Y4.8899
G1 X-5.3915 Y4.8881
G3 X-5.4996 Y4.8257 I0.0001 J-0.125
G1 X-6.9785 Y2.2675
G3 X-6.9953 Y2.2049 I0.1082 J-0.0626
G3 X-6.9785 Y2.1424 I0.125 J0
G1 X-5.7791 Y0.0652
G1 X-6.9912 Y-2.0343
G3 X-7.0079 Y-2.0968 I0.1083 J-0.0625
G3 X-6.9912 Y-2.1593 I0.125 J0
G1 X-5.5121 Y-4.7238
G3 X-5.4039 Y-4.7863 I0.1083 J0.0625
G1 X-3.0014 Y-4.7876
G1 X-1.7718 Y-6.8577
G3 X-1.6655 Y-6.9189 I0.1075 J0.0638
G1 X1.2743 Y-6.9463
G3 X1.3837 Y-6.8838 I0.0012 J0.125
G1 X2.583 Y-4.8064
G1 X5.01 Y-4.8046
G3 X5.1182 Y-4.7421 I-0.0001 J0.125
G1 X6.5969 Y-2.1807
G3 X6.6136 Y-2.1182 I-0.1083 J0.0625
G3 X6.5968 Y-2.0556 I-0.125 J0
G1 X5.3977 Y0.0184
G1 X6.6097 Y2.1179
G3 X6.6265 Y2.1804 I-0.1083 J0.0625
G3 X6.6098 Y2.2428 I-0.125 J0
G1 X5.1336 Y4.8058
G3 X5.0253 Y4.8684 I-0.1083 J-0.0624
G1 X2.6266 Y4.8684
G1 X1.4144 Y6.9678
G3 X1.3062 Y7.0303 I-0.1083 J-0.0625
G1 Z-0.25 F30
G1 X-1.6542 Y7.0316 F15
G3 X-1.7625 Y6.9692 I-0.0001 J-0.125
G1 X-2.9644 Y4.8899
G1 X-5.3915 Y4.8881
G3 X-5.4996 Y4.8257 I0.0001 J-0.125
G1 X-6.9785 Y2.2675
G3 X-6.9953 Y2.2049 I0.1082 J-0.0626
G3 X-6.9785 Y2.1424 I0.125 J0
G1 X-5.7791 Y0.0652
G1 X-6.9912 Y-2.0343
G3 X-7.0079 Y-2.0968 I0.1083 J-0.0625
G3 X-6.9912 Y-2.1593 I0.125 J0
G1 X-5.5121 Y-4.7238
G3 X-5.4039 Y-4.7863 I0.1083 J0.0625
G1 X-3.0014 Y-4.7876
G1 X-1.7718 Y-6.8577
G3 X-1.6655 Y-6.9189 I0.1075 J0.0638
G1 X1.2743 Y-6.9463
G3 X1.3837 Y-6.8838 I0.0012 J0.125
G1 X2.583 Y-4.8064
G1 X5.01 Y-4.8046
G3 X5.1182 Y-4.7421 I-0.0001 J0.125
G1 X6.5969 Y-2.1807
G3 X6.6136 Y-2.1182 I-0.1083 J0.0625
G3 X6.5968 Y-2.0556 I-0.125 J0
G1 X5.3977 Y0.0184
G1 X6.6097 Y2.1179
G3 X6.6265 Y2.1804 I-0.1083 J0.0625
G3 X6.6098 Y2.2428 I-0.125 J0
G1 X5.1336 Y4.8058
G3 X5.0253 Y4.8684 I-0.1083 J-0.0624
G1 X2.6266 Y4.8684
G1 X1.4144 Y6.9678
G3 X1.3062 Y7.0303 I-0.1083 J-0.0625
G1 Z-0.3 F30
G1 X-1.6542 Y7.0316 F15
G3 X-1.7625 Y6.9692 I-0.0001 J-0.125
G1 X-2.9644 Y4.8899
G1 X-5.3915 Y4.8881
G3 X-5.4996 Y4.8257 I0.0001 J-0.125
G1 X-6.9785 Y2.2675
G3 X-6.9953 Y2.2049 I0.1082 J-0.0626
G3 X-6.9785 Y2.1424 I0.125 J0
G1 X-5.7791 Y0.0652
G1 X-6.9912 Y-2.0343
G3 X-7.0079 Y-2.0968 I0.1083 J-0.0625
G3 X-6.9912 Y-2.1593 I0.125 J0
G1 X-5.5121 Y-4.7238
G3 X-5.4039 Y-4.7863 I0.1083 J0.0625
G1 X-3.0014 Y-4.7876
G1 X-1.7718 Y-6.8577
G3 X-1.6655 Y-6.9189 I0.1075 J0.0638
G1 X1.2743 Y-6.9463
G3 X1.3837 Y-6.8838 I0.0012 J0.125
G1 X2.583 Y-4.8064
G1 X5.01 Y-4.8046
G3 X5.1182 Y-4.7421 I-0.0001 J0.125
G1 X6.5969 Y-2.1807
G3 X6.6136 Y-2.1182 I-0.1083 J0.0625
G3 X6.5968 Y-2.0556 I-0.125 J0
G1 X5.3977 Y0.0184
G1 X6.6097 Y2.1179
G3 X6.6265 Y2.1804 I-0.1083 J0.0625
G3 X6.6098 Y2.2428 I-0.125 J0
G1 X5.1336 Y4.8058
G3 X5.0253 Y4.8684 I-0.1083 J-0.0624
G1 X2.6266 Y4.8684
G1 X1.4144 Y6.9678
G3 X1.3062 Y7.0303 I-0.1083 J-0.0625
G1 Z-0.35 F30
G1 X-1.6542 Y7.0316 F15
G3 X-1.7625 Y6.9692 I-0.0001 J-0.125
G1 X-2.9644 Y4.8899
G1 X-5.3915 Y4.8881
G3 X-5.4996 Y4.8257 I0.0001 J-0.125
G1 X-6.9785 Y2.2675
G3 X-6.9953 Y2.2049 I0.1082 J-0.0626
G3 X-6.9785 Y2.1424 I0.125 J0
G1 X-5.7791 Y0.0652
G1 X-6.9912 Y-2.0343
G3 X-7.0079 Y-2.0968 I0.1083 J-0.0625
G3 X-6.9912 Y-2.1593 I0.125 J0
G1 X-5.5121 Y-4.7238
G3 X-5.4039 Y-4.7863 I0.1083 J0.0625
G1 X-3.0014 Y-4.7876
G1 X-1.7718 Y-6.8577
G3 X-1.6655 Y-6.9189 I0.1075 J0.0638
G1 X1.2743 Y-6.9463
G3 X1.3837 Y-6.8838 I0.0012 J0.125
G1 X2.583 Y-4.8064
G1 X5.01 Y-4.8046
G3 X5.1182 Y-4.7421 I-0.0001 J0.125
G1 X6.5969 Y-2.1807
G3 X6.6136 Y-2.1182 I-0.1083 J0.0625
G3 X6.5968 Y-2.0556 I-0.125 J0
G1 X5.3977 Y0.0184
G1 X6.6097 Y2.1179
G3 X6.6265 Y2.1804 I-0.1083 J0.0625
G3 X6.6098 Y2.2428 I-0.125 J0
G1 X5.1336 Y4.8058
G3 X5.0253 Y4.8684 I-0.1083 J-0.0624
G1 X2.6266 Y4.8684
G1 X1.4144 Y6.9678
G3 X1.3062 Y7.0303 I-0.1083 J-0.0625
G1 Z-0.4 F30
G1 X-1.6542 Y7.0316 F15
G3 X-1.7625 Y6.9692 I-0.0001 J-0.125
G1 X-2.9644 Y4.8899
G1 X-5.3915 Y4.8881
G3 X-5.4996 Y4.8257 I0.0001 J-0.125
G1 X-6.9785 Y2.2675
G3 X-6.9953 Y2.2049 I0.1082 J-0.0626
G3 X-6.9785 Y2.1424 I0.125 J0
G1 X-5.7791 Y0.0652
G1 X-6.9912 Y-2.0343
G3 X-7.0079 Y-2.0968 I0.1083 J-0.0625
G3 X-6.9912 Y-2.1593 I0.125 J0
G1 X-5.5121 Y-4.7238
G3 X-5.4039 Y-4.7863 I0.1083 J0.0625
G1 X-3.0014 Y-4.7876
G1 X-1.7718 Y-6.8577
G3 X-1.6655 Y-6.9189 I0.1075 J0.0638
G1 X1.2743 Y-6.9463
G3 X1.3837 Y-6.8838 I0.0012 J0.125
G1 X2.583 Y-4.8064
G1 X5.01 Y-4.8046
G3 X5.1182 Y-4.7421 I-0.0001 J0.125
G1 X6.5969 Y-2.1807
G3 X6.6136 Y-2.1182 I-0.1083 J0.0625
G3 X6.5968 Y-2.0556 I-0.125 J0
G1 X5.3977 Y0.0184
G1 X6.6097 Y2.1179
G3 X6.6265 Y2.1804 I-0.1083 J0.0625
G3 X6.6098 Y2.2428 I-0.125 J0
G1 X5.1336 Y4.8058
G3 X5.0253 Y4.8684 I-0.1083 J-0.0624
G1 X2.6266 Y4.8684
G1 X1.4144 Y6.9678
G3 X1.3062 Y7.0303 I-0.1083 J-0.0625
G1 Z-0.45 F30
G1 X-1.6542 Y7.0316 F15
G3 X-1.7625 Y6.9692 I-0.0001 J-0.125
G1 X-2.9644 Y4.8899
G1 X-5.3915 Y4.8881
G3 X-5.4996 Y4.8257 I0.0001 J-0.125
G1 X-6.9785 Y2.2675
G3 X-6.9953 Y2.2049 I0.1082 J-0.0626
G3 X-6.9785 Y2.1424 I0.125 J0
G1 X-5.7791 Y0.0652
G1 X-6.9912 Y-2.0343
G3 X-7.0079 Y-2.0968 I0.1083 J-0.0625
G3 X-6.9912 Y-2.1593 I0.125 J0
G1 X-5.5121 Y-4.7238
G3 X-5.4039 Y-4.7863 I0.1083 J0.0625
G1 X-3.0014 Y-4.7876
G1 X-1.7718 Y-6.8577
G3 X-1.6655 Y-6.9189 I0.1075 J0.0638
G1 X1.2743 Y-6.9463
G3 X1.3837 Y-6.8838 I0.0012 J0.125
G1 X2.583 Y-4.8064
G1 X5.01 Y-4.8046
G3 X5.1182 Y-4.7421 I-0.0001 J0.125
G1 X6.5969 Y-2.1807
G3 X6.6136 Y-2.1182 I-0.1083 J0.0625
G3 X6.5968 Y-2.0556 I-0.125 J0
G1 X5.3977 Y0.0184
G1 X6.6097 Y2.1179
G3 X6.6265 Y2.1804 I-0.1083 J0.0625
G3 X6.6098 Y2.2428 I-0.125 J0
G1 X5.1336 Y4.8058
G3 X5.0253 Y4.8684 I-0.1083 J-0.0624
G1 X2.6266 Y4.8684
G1 X1.4144 Y6.9678
G3 X1.3062 Y7.0303 I-0.1083 J-0.0625
G1 Z-0.5 F30
G1 X-1.6542 Y7.0316 F15
G3 X-1.7625 Y6.9692 I-0.0001 J-0.125
G1 X-2.9644 Y4.8899
G1 X-5.3915 Y4.8881
G3 X-5.4996 Y4.8257 I0.0001 J-0.125
G1 X-6.9785 Y2.2675
G3 X-6.9953 Y2.2049 I0.1082 J-0.0626
G3 X-6.9785 Y2.1424 I0.125 J0
G1 X-5.7791 Y0.0652
G1 X-6.9912 Y-2.0343
G3 X-7.0079 Y-2.0968 I0.1083 J-0.0625
G3 X-6.9912 Y-2.1593 I0.125 J0
G1 X-5.5121 Y-4.7238
G3 X-5.4039 Y-4.7863 I0.1083 J0.0625
G1 X-3.0014 Y-4.7876
G1 X-1.7718 Y-6.8577
G3 X-1.6655 Y-6.9189 I0.1075 J0.0638
G1 X1.2743 Y-6.9463
G3 X1.3837 Y-6.8838 I0.0012 J0.125
G1 X2.583 Y-4.8064
G1 X5.01 Y-4.8046
G3 X5.1182 Y-4.7421 I-0.0001 J0.125
G1 X6.5969 Y-2.1807
G3 X6.6136 Y-2.1182 I-0.1083 J0.0625
G3 X6.5968 Y-2.0556 I-0.125 J0
G1 X5.3977 Y0.0184
G1 X6.6097 Y2.1179
G3 X6.6265 Y2.1804 I-0.1083 J0.0625
G3 X6.6098 Y2.2428 I-0.125 J0
G1 X5.1336 Y4.8058
G3 X5.0253 Y4.8684 I-0.1083 J-0.0624
G1 X2.6266 Y4.8684
G1 X1.4144 Y6.9678
G3 X1.3062 Y7.0303 I-0.1083 J-0.0625
G1 Z-0.55 F30
G1 X-1.6542 Y7.0316 F15
G3 X-1.7625 Y6.9692 I-0.0001 J-0.125
G1 X-2.9644 Y4.8899
G1 X-5.3915 Y4.8881
G3 X-5.4996 Y4.8257 I0.0001 J-0.125
G1 X-6.9785 Y2.2675
G3 X-6.9953 Y2.2049 I0.1082 J-0.0626
G3 X-6.9785 Y2.1424 I0.125 J0
G1 X-5.7791 Y0.0652
G1 X-6.9912 Y-2.0343
G3 X-7.0079 Y-2.0968 I0.1083 J-0.0625
G3 X-6.9912 Y-2.1593 I0.125 J0
G1 X-5.5121 Y-4.7238
G3 X-5.4039 Y-4.7863 I0.1083 J0.0625
G1 X-3.0014 Y-4.7876
G1 X-1.7718 Y-6.8577
G3 X-1.6655 Y-6.9189 I0.1075 J0.0638
G1 X1.2743 Y-6.9463
G3 X1.3837 Y-6.8838 I0.0012 J0.125
G1 X2.583 Y-4.8064
G1 X5.01 Y-4.8046
G3 X5.1182 Y-4.7421 I-0.0001 J0.125
G1 X6.5969 Y-2.1807
G3 X6.6136 Y-2.1182 I-0.1083 J0.0625
G3 X6.5968 Y-2.0556 I-0.125 J0
G1 X5.3977 Y0.0184
G1 X6.6097 Y2.1179
G3 X6.6265 Y2.1804 I-0.1083 J0.0625
G3 X6.6098 Y2.2428 I-0.125 J0
G1 X5.1336 Y4.8058
G3 X5.0253 Y4.8684 I-0.1083 J-0.0624
G1 X2.6266 Y4.8684
G1 X1.4144 Y6.9678
G3 X1.3062 Y7.0303 I-0.1083 J-0.0625
G0 Z0.25
M5
M30
`

**Comment by johnbenweber** [comment]
Good catch, I'll have a look. If you have a collection of gcode files you've found to be tricky it would be good to include those in the GcodeForTesting directory so we can test changes against a consistent set of gcode files.

**Comment by BarbourSmith** [comment]
That's a good idea, I'll add more right now.

**Comment by johnbenweber** [comment]
I think I fixed it. Looks like there was a typo ("maht.pi") in the calculation of angles in quadrant three.

**Comment by BarbourSmith** [comment]
Looks great! Good catch 👍

**Pull request merged**